default/methods/source_control/file/file_browser.py

```diff
@@ -372,6 +372,54 @@ def get_instance_buffer(session, video_file_id, start, end):
     return buffer
 
 
+@routes.route('/api/project/<string:project_string_id>/files', methods = ['GET'])
+@Project_permissions.user_has_project([
+    "admin", "Editor", "Viewer", "allow_if_project_is_public"])
+def get_files(project_string_id):
+
+    with sessionMaker.session_scope() as session:
+
+        project = Project.get(session, project_string_id)
+
+        data = request.get_json(force = True)
+        query = request.args.get('query')
+
+        metadata_proposed = data.get('metadata')
+
+        if not metadata_proposed:
+            return jsonify("Error no metadata"), 400
+
+        # Add query to the proposed metadata so that File_Browser doesnt need to be changed
+        metadata_proposed['query'] = query
```

Comment:
To avoid changes to `File_Browser`, I simply added the query field to `metadata_proposed`. 

Wasnt sure how much of the `metadata_proposed` and `File_Browser` we want to change for the moment, so left them mostly intact for now